custom_components/mypyllant/utils.py

```diff
@@ -12,10 +13,15 @@ def get_system_sensor_unique_id(system: System) -> str:
     Otherwise, we fall back to the system id
     """
     if system.primary_heat_generator:
-        return system.primary_heat_generator.device_uuid
+        return f"{system.id}_{system.primary_heat_generator.device_uuid}"
     else:
         return system.id
 
+def get_unique_id_prefix(system_id: str) -> str:
+    return f"{DOMAIN}_{system_id}_"
+
+def get_name_prefix(home: str) -> str:
```

Comment:
I think you still need to install pre-commit and run it on your changes. I've merged your other PR and updated the pre-commit settings and VS Code lint task.